cla_frontend/assets-src/javascripts/app/partials/csv_download_list.html

```diff
@@ -2,6 +2,16 @@
   <h2>CSV Download</h2>
 </header>
 
+<form class="FormRow" name="csvDownloadFilters" ng-submit="submitFilters()">
+  <label>
+    Filter by provider:
+    <select name="provider" ng-model="provider" ng-options="p.id as p.name for p in providers | orderBy:'name'">
+      <option value="">(No filter)</option>
+    </select>
+  </label>
+  <button type="submit" class="Button Button--secondary" ng-disabled="csvDownloadFilters.$invalid">Filter</button>
```

Comment:
I think it would be better to apply filter on change event to avoid extra clicks on sole action.